### ISSUE TITLE
Black screen after received push message and doubleclicked to activate

### ISSUE THREAD

**Issue by shh0** [siacs/Conversations#3467]
#### General information

* **Version:** 2.5.1+fcr
* **Device:** Xiaomi Mi A1
* **Android Version:** Android 9 Stock (1.Mai)
* **Server name:** self hosted
* **Server software:** prosody 0.11-2
* **Installed server modules:**  identical to conversations.im
* **Conversations source:** F-Droid

#### Steps to reproduce
1. Android is in sleep mode (screen is black)
2. Receive a (push?) message.
3. Message is shown on the screen.
4. Doubleclick on message.  (choosing "Antworten/Answer" works as espected!)

#### Expected result
Show Conversations and editing window.

#### Actual result
Black screen (of Conversations). Title bar is still visible (and reactive) though.

**Comment by licaon-kter** [comment]
No Push on F-Droid, so that's a normal message.

Logcat when this happens?

**Comment by shh0** [comment]
> No Push on F-Droid, so that's a normal message.

Ah, okay.

> Logcat when this happens?

This happens all the time. But, I'll try to get a log.…….

**Comment by shh0** [comment]
Finally got a logcat: [logcat.txt](https://github.com/siacs/Conversations/files/3316772/logcat.txt)

The idle phone received one message while screen was off.
Msg was shown properly on the screen. Then I double-clicked on the message. Result: (just) a black Conversations window...

**Comment by licaon-kter** [comment]
On mobile you touch...why do you "double click"?

**Comment by shh0** [comment]
Ok, I was double-touching... ;)

**Comment by licaon-kter** [comment]
By "shown on screen" you mean on `Lock screen` ?

**Comment by shh0** [comment]
Yes!

**Comment by licaon-kter** [comment]
Do you have a PIN/password/fingerprint? Is the notification showing the message text in full or hidden?

**Comment by shh0** [comment]
> Do you have a PIN/password/fingerprint?

No! I don't have any locking mechanism activated.

>  Is the notification showing the message text in full or hidden?

[screen get's activated]. Full message is shown. [screen get's deactivated after a few seconds if I don't do anything]

> Did you disable battery optimisations for Conversations?

I don't think so. Where is this option? 

 > Did you enable the Expert ->Foreground service?

No. The checkbox isn't checked. (so Conversations isn't running in foreground mode I assume)

> Do you have any "optimizer" apps, eg. Xposed/Magisk modules?

No. That's just a "pure" Android 9. However, I disabled Googles "Intelligent Battery" option.

> Same behaviour on 2.5.2?

Yes. Currently I'm running v.2.5.2

**Comment by licaon-kter** [comment]
> I don't think so. Where is this option?

Android Settings -> Apps -> Special permissions -> Ignore battery optimizations -> All apps -> Conversations ( do not optimize) 

_(FYI, you have been asked this at one point, maybe it's already ignored)_

> No. The checkbox isn't checked. (so Conversations isn't running in foreground mode I assume)

Mmm, why not? F-Droid builds don't benefit from Push, so you need this so Android does not kill the app. If you want to hide the useless "X accounts connected" notification read here: https://github.com/siacs/Conversations#im-getting-this-annoying-permanent-notification

Do these....retest...

**Comment by shh0** [comment]
> Android Settings -> Apps -> Special permissions -> Ignore battery optimizations -> All apps -> Conversations ( do not optimize)

Status of this was already at "do not optimize".

> Mmm, why not? F-Droid builds don't benefit from Push, so you need this so Android does not kill the app. If you want to hide the useless "X accounts connected" notification read here: https://github.com/siacs/Conversations#im-getting-this-annoying-permanent-notification

Mmm, I thought Android wouldn't kill the app anyways if battery optimizing option is at "do not optimize"?

However, activating the foreground mode didn't change anything. On double-tabbing/touching, Conversations somehow seems to start for editing, but shows a completely black window.
So, the double-tabbing isn't connected to the method correctly, I think.
Does Android 9 have a new API-method for this lock screen messages which has to be implemented somehow? E.g. default method for double-tabbing should be "Answer" instead of *null*... [just guessing]

**Comment by licaon-kter** [comment]
Why do you "double"-tap, is that supposed to unlock and open?

What if you unlock and then touch? Unlock and double-touch?

Can you test with a PIN/password/fingerprint?

**Comment by shh0** [comment]
> Why do you "double"-tap, is that supposed to unlock and open?

No. 
Tapping once is just selecting the message. Android then writes "tap once again to open the message" (or something similar; my phone is talking German)
Therefore I'm double tapping (or tapping twice) to open the message (and the related app). For WhatsApp it's working the same way.

> What if you unlock and then touch?
> Unlock and double-touch?

The phone is always unlocked. Un"locking" would generally be to push the power button. But after receiving a message (where the screen get's on), the button just switches the screen off immediately.
Touching the message once just selects/highlights the msg.

> Can you test with a PIN/password/fingerprint?

Will check.... ... ...

**Comment by shh0** [comment]
> Can you test with a PIN/password/fingerprint?

Checked. Works now as expected that way: No black screen.

But, I now also checked WhatsApp with my previous config without a PIN or  other locking method:
WhatsApp shows me a black screen after double-tap, too!

So, my Xiaomi Android One 9 Pie version seems to be broken somehow when configured without a locking method. And Conversations is innocent, sorry for the hassle!
**Thank you very much** for your help!
